corehq/apps/domain/templates/domain/stripe_cards.html

```diff
@@ -0,0 +1,93 @@
+{% load hq_shared_tags %}
+{% load i18n %}
+
+<fieldset>
+    <legend>{% trans 'Saved Credit Cards' %}</legend>
+    <table class='table'>
+        <thead>
+            <tr>
+                <th style="width:20%;"></th>
+                <th style="width:20%;"></th>
+                <th style="width:20%;"></th>
+                <th style="width:20%;"></th>
+                <th style="width:20%;"></th>
+            </tr>
+        </thead>
+        <tbody>
+            <!-- ko foreach: cards -->
+                <tr>
+                    <td>
+                        <i class="icon-credit-card"></i> <strong><span data-bind="text: brand"></span></strong>
+                    </td>
+                    <td>
+                        ************<strong><span data-bind="text: last4"></span></strong>
+                    </td>
+                    <td><span data-bind="text: exp_month"></span>/<span data-bind="text: exp_year"></span></td>
+                    <td>
+                        <span class="label label-success"
+                              data-bind="visible: is_autopay">
+                            {% trans 'Autopay card' %}
+                        </span>
+                        <button data-bind="visible: is_autopay,
+                                           click: unSetAutopay,
+                                           enable: $parent.autoPayButtonEnabled"
+                                class='btn btn-danger autopay-button'>
+                            {% trans 'Remove' %}
+                        </button>
+                        <button data-bind="visible: !is_autopay(),
+                                           click: setAutopay,
+                                           enable: $parent.autoPayButtonEnabled"
+                                class='btn autopay-button'>
+                            {% trans 'Set as autopay card' %}
+                        </button>
+                    </td>
+                    <td>
+                        <button data-bind="click: toggleDeleteConfirmation,
+                                           visible: !showDeleteConfirmation()"
+                                class="btn btn-danger">
+                            <i class="icon-remove"></i> {% trans 'Delete Card' %}
+                        </button>
+                        <div class="alert"
+                             style="margin-bottom:0;"
```

Comment:
thanks jenny that's definitely the better answer

On Mon, Oct 5, 2015 at 4:33 PM, Jenny Schweers <notifications@github.com>
wrote:

> In corehq/apps/domain/templates/domain/stripe_cards.html
> <https://github.com/dimagi/commcare-hq/pull/8138#discussion_r41193771>:
>
> > +                        </button>
> > +                        <button data-bind="visible: !is_autopay(),
> > +                                           click: setAutopay,
> > +                                           enable: $parent.autoPayButtonEnabled"
> > +                                class='btn autopay-button'>
> > +                            {% trans 'Set as autopay card' %}
> > +                        </button>
> > +                    </td>
> > +                    <td>
> > +                        <button data-bind="click: toggleDeleteConfirmation,
> > +                                           visible: !showDeleteConfirmation()"
> > +                                class="btn btn-danger">
> > +                            <i class="icon-remove"></i> {% trans 'Delete Card' %}
> > +                        </button>
> > +                        <div class="alert"
> > +                             style="margin-bottom:0;"
>
> Hey, the underlying problem here is that the alert classes are meant for
> page-level information, not for inline use. The usual pattern for
> confirming an action like this is to pop up a modal - check out how the web
> users page (https://www.commcarehq.org/a/jennytraining/settings/users/web/)
> handles deletions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8138/files#r41193771>.
>